### PR TITLE
Add rl/height-weight-analysis into spine-generic:master

### PR DESCRIPTION
**Pull request by renelabounek**
Hi @jcohenadad, here is the requested PR. I am still working on help/documentation texts, but the code might be ready.
If you would like to replicate the graphs and results, pseudocode of 4 necessary steps would be as follows:

# 1. process_data.sh
Extended list of outputs
extract_measures.sh file might be deleted as everything is included in the updated process_data.sh

# 2. Download:
[fs_measurements.xlsx](https://github.com/umn-milab/spine-generic-body-size-results/blob/main/fs-measurements.xlsx), [sg.lh.aparc.stats.postcentral.csv](https://github.com/umn-milab/spine-generic-body-size-results/blob/main/sg.lh.aparc.stats.postcentral.csv), [sg.lh.aparc.stats.precentral.csv](https://github.com/umn-milab/spine-generic-body-size-results/blob/main/sg.lh.aparc.stats.precentral.csv), [sg.rh.aparc.stats.postcentral.csv](https://github.com/umn-milab/spine-generic-body-size-results/blob/main/sg.rh.aparc.stats.postcentral.csv) and [sg.rh.aparc.stats.precentral.csv](https://github.com/umn-milab/spine-generic-body-size-results/blob/main/sg.rh.aparc.stats.precentral.csv) files into the ${PATH_RESULTS} folder
   - fs_measureemnts.xlsx table was provided by @mbondy023 who visually inspected each segmentation prior adding numbers of each participant into the table
   - .csv files were additionally generated from FS result files with extract_fs_measures.sh script in the root folder (after the @mbondy023's visual inspection)
   - Pitfall: FS pipeline is not automated into the spine-generic pipeline now. Do we want and need to improve this? UMN preference was to rahter share just the table files, as FS is stochastic and leading to slightly different numbers per each run.
   - BTW all my [${PATH_RESULTS}](https://github.com/umn-milab/spine-generic-body-size-results) including the 4 above listed same files

# 3. Open MATLAB and execute:

```
cd('~/spine-generic/matlab')
stat = sg_structure_versus_demography(path_results,path_data)
```
In bash pseudocode you need to execute the function with following settings:
```
cd('`which process_data.sh`/matlab')
stat = sg_structure_versus_demography('${PATH_RESULTS}','${PATH_DATA}')
```
PATH_DATA is necessary to find the ``exclude.yml`` file which is not copied into the PATH_RESULTS folder during the process_data.sh
I tested MATLAB code on version R2021b, because MathWorks introduced quite inteligent automated axis label orientation and saved me time and nerves with it... :-) 

# 4. Bunch of fig_*.png files should appear in the ${PATH_RESULTS} folder.
From the matlab stat variable, you might be interested in stat.tbl

#
Please let me know, if something is not clear.
Good luck,
Rene

### CODE REVIEW

**Review by jcohenadad — COMMENTED** [review]
we probably don't need these:
- matlab/Tests/Data/test_import/~temporary.yaml
- matlab/Tests/Data/test_import/selftest_report.html
- matlab/Tests/selftest_report.html

and other temporary and non-code data.

**Comment by renelabounek** [comment]
> we probably don't need these:
> 
> * matlab/Tests/Data/test_import/~temporary.yaml
> * matlab/Tests/Data/test_import/selftest_report.html
> * matlab/Tests/selftest_report.html
> 
> and other temporary and non-code data.

@jcohenadad Yes, you are right some files from the Tests folder might be deleted. I tried to delete the whole Tests folder but the [yamlmatlab](https://code.google.com/archive/p/yamlmatlab/downloads) toolbox stopped to work then. I will try the old-school Windows 95 optimization method, alá: *"Try everything what can be deleted from the C:\WINDOWS and the system is still working..."*

But note, that after deleting of these files we will not have in the code the exact copy of the [yamlmatlab](https://code.google.com/archive/p/yamlmatlab/downloads) toolbox.

**Comment by jcohenadad** [comment]
> But note, that after deleting of these files we will not have in the code the exact copy of the [yamlmatlab](https://code.google.com/archive/p/yamlmatlab/downloads) toolbox

how about adding as a dependency this toolbox (with its version) in the README for people who would like to reproduce these results?  The philosophy is not to include the code of others (which could conflict with our license), but to instruct how to download it

**Comment by renelabounek** [comment]
> how about adding as a dependency this toolbox (with its version) in the README for people who would like to reproduce these results? The philosophy is not to include the code of others (which could conflict with our license), but to instruct how to download it

Yes, we can do it this way, not a big deal.  I will delete it and include description and URL link into README.md

Originally, I thought that as the [yamlmatlab](https://code.google.com/archive/p/yamlmatlab/downloads) toolbox is under [MIT license](https://github.com/renelabounek/spine-generic/blob/rl/height-weight-analysis/matlab/YAMLMatlab_0.4.3_MIT-license.txt), it is fine to extend it into our source code when their copyright/license is acknowledged. I also thought that ANTs registration is similarly included into the SCT.

**Comment by jcohenadad** [comment]
> > how about adding as a dependency this toolbox (with its version) in the README for people who would like to reproduce these results? The philosophy is not to include the code of others (which could conflict with our license), but to instruct how to download it
> 
> Yes, we can do it this way, not a big deal. I will delete it and include description and URL link into README.md
> 
> Originally, I thought that as the [yamlmatlab](https://code.google.com/archive/p/yamlmatlab/downloads) toolbox is under [MIT license](https://github.com/renelabounek/spine-generic/blob/rl/height-weight-analysis/matlab/YAMLMatlab_0.4.3_MIT-license.txt), it is fine to extend it into our source code when their copyright/license is acknowledged. 

Indeed, the MIT license would be compatible so it should be fine to include it, but good practice is to mention external dependencies instead of including them.

> I also thought that ANTs registration is similarly included into the SCT.

Actually, ANTs code is not included in SCT's repository but it is downloaded upon installation of SCT by the install script. So this is exactly what I am suggesting to you here: having instructions (or installation script if you package it) that fetches a specific version of YAMLMatlab to install it upon installation of the spine-generic project. What complicates things here, however, is that we are dealing with two different languages (python and Matlab), and packaging becomes a bit of a nightmare. Which is why instructions (vs. install script) seems to me the most reasonable approach.

**Comment by jcohenadad** [comment]
> How will we name the current analysis in the documentation? "Body size versus the structure of the central nervous system" ? For headings, sub-headings, references, etc.

yup! sounds good

> "Fetch" sounds me like somethin github related. I am not sure if it is possible for the yamlmatlab toolbox as only place where I found it stored is google archive: https://code.google.com/archive/p/yamlmatlab/downloads

Sorry I'm not sure I understand the question. My suggestion was to write in the instructions something like this:

Section: Body size versus the structure of the central nervous system

Dependencies: [YAMLMatlab_0.4.3](https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/yamlmatlab/YAMLMatlab_0.4.3.zip)

**Comment by renelabounek** [comment]
@jcohenadad UPDATE:

- New file: [matlab/sg_example_script.m](https://github.com/renelabounek/spine-generic/blob/rl/height-weight-analysis/matlab/sg_example_script.m) explaining how to set YAMLMatlab toolbox and other variables
- Deleted YAMLMatlab toolbox from the source code
- New doc file: [docs/body-size-pipeline.rst](https://github.com/renelabounek/spine-generic/blob/rl/height-weight-analysis/docs/body-size-pipeline.rst) explaining how to replicate Results (I have shortened the title due to limited space in the website left sidebar)
-  Edited: [docs/index.rst](https://github.com/renelabounek/spine-generic/blob/rl/height-weight-analysis/docs/index.rst)

Please let me know, what else needs to be revised. I hope the updated docs stayed compatible and functional.

File: docs/body-size-pipeline.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+--------------------------
+
+The same methods as used in the `Analysis pipeline <https://spine-generic.readthedocs.io/analysis-pipeline.html>`__, i.e sub-sections `Getting started <https://spine-generic.readthedocs.io/analysis-pipeline.html#getting-started>`__ and `Quality Control <https://spine-generic.readthedocs.io/analysis-pipeline.html#quality-control>`__.
+
+.. _cerebral-image-analysis:
+
+Cerebral image analysis
+-----------------------
+
+You need to proceed your own automated cerebral segmentation of **T1** scans through the `FreeSurfer >=7.2 <https://surfer.nmr.mgh.harvard.edu>`__ software or download values obtained by the `UMN MILab <https://github.com/umn-milab>`__, which visually inspected and corrected (if necessary) all segmentation outcomes prior to exttraction of the quantitative measurements.
```
Comment (jcohenadad) [inline docs/body-size-pipeline.rst:35]:
```suggestion
You need to proceed your own automated cerebral segmentation of **T1** scans through the `FreeSurfer >=7.2 <https://surfer.nmr.mgh.harvard.edu>`__ software or download values obtained by the `UMN MILab <https://github.com/umn-milab>`__, which visually inspected and corrected (if necessary) all segmentation outcomes prior to extraction of the quantitative measurements.
```

File: docs/body-size-pipeline.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+.. code-block:: bash
+
+    git clone https://github.com/umn-milab/spine-generic-body-size-results.git
+    cd spine-generic-body-size-results
+    cp fs_measurements.xlsx sg.*.aparc.stats.*.csv <PATH_RESULTS>
+
+Your own FreeSurfer segmentation and measurements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+-   Group ``brain_t1`` in the `exclude.yml <https://github.com/spine-generic/data-multi-subject/blob/master/exclude.yml>`__ file defines list of scans whose cerebral volumes are not possible to be sgemented correctly due to listed reasons.
```
Comment (jcohenadad) [inline docs/body-size-pipeline.rst:49]:
```suggestion
-   Group ``brain_t1`` in the `exclude.yml <https://github.com/spine-generic/data-multi-subject/blob/master/exclude.yml>`__ file defines list of scans whose cerebral volumes are not possible to be segmented correctly due to listed reasons.
```

File: docs/body-size-pipeline.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    stat = sg_structure_versus_demography(path_results,path_data);
+    % Figures exported in fig_*.png format and stat_labounek2022.mat file
+    % will appear in the <PATH_RESULTS> folder
+
+.. _results:
+
+Results
+-------
+
+(to be extended when whole dataset analyzed)
```
Comment (jcohenadad) [inline docs/body-size-pipeline.rst:85]:
```suggestion
(to be extended once the whole dataset is analyzed)
```

**Review by jcohenadad — COMMENTED** [review]
Looks good @renelabounek ! I only made minor suggestions. It can be merged afterwards

**Comment by renelabounek** [comment]
> Looks good @renelabounek ! I only made minor suggestions. It can be merged afterwards

@Igor-Nestrasil has also approved final code and docs version. So @jcohenadad, you can approve the review and let the branch to be merged. I do not have the right to approve the review.
Thanks, Rene

**Review by renelabounek — COMMENTED** [review]
Everything looks fine. I can not Approve the review.

**Review by jcohenadad — APPROVED** [review]

**Comment by renelabounek** [comment]
@jcohenadad Can you please force merge of this PR? Again, it stopped on the build python checks. Like previously here https://github.com/spine-generic/spine-generic/pull/253#issuecomment-1005115882 and here https://github.com/spine-generic/spine-generic/pull/255#issuecomment-989120707
I have no  clue why it is happening when I have commited zero to the python code. It seems that you did not also identified the cause before.

**Pull request merged**